libcontainer/cgroups/systemd/v2.go

```diff
@@ -477,3 +477,37 @@ func (m *UnifiedManager) Exists() bool {
 func (m *UnifiedManager) OOMKillCount() (uint64, error) {
 	return m.fsMgr.OOMKillCount()
 }
+
+func (m *UnifiedManager) MakeThreaded() error {
+	const (
+		cgTypeFile   = "cgroup.type"
+		typeThreaded = "threaded"
+	)
+
+	cgroupType, err := cgroups.ReadFile(m.path, cgTypeFile)
+	if err != nil {
+		return err
+	}
+	cgroupType = strings.TrimSpace(cgroupType)
+
+	if cgroupType == typeThreaded {
+		return nil
+	}
+
+	err = cgroups.WriteFile(m.path, cgTypeFile, typeThreaded)
+	if err != nil {
+		return err
+	}
+
+	cgroupType, err = cgroups.ReadFile(m.path, cgTypeFile)
+	if err != nil {
+		return err
+	}
+	cgroupType = strings.TrimSpace(cgroupType)
+
+	if cgroupType != typeThreaded {
+		return fmt.Errorf("could not change cgroup type (%s) to %s", cgroupType, typeThreaded)
+	}
+
+	return nil
```

Comment:
Can't we call `m.fsMgr.MakeThreaded()` instead of the copy paste?

Reply:
Yes, absolutely!
Done.